fix(community): don't delete chat and messages when leaving community

jrainville opened the pull request:
Desktop issue: https://github.com/status-im/status-desktop/issues/7512

This creates a smoother experience for users when they leave a community since they can see the exact same messages they had before without having to rely on the mailserver.

To make it work on the desktop or mobile, side you need to make sure that you re-fetch the messages for the channels when re-joining (or re-spectating). This might require to reset the cursor.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (9)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ceb6c051 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3006/1/) | 2022-12-06 20:12:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221206-201107-ceb6c0-pr3006.zip) |
| :heavy_check_mark: | ceb6c051 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3006/1/) | 2022-12-06 20:15:26 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221206-201102-ceb6c0-pr3006.aar) |
| :heavy_check_mark: | ceb6c051 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3006/1/) | 2022-12-06 20:16:17 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221206-201102-ceb6c0-pr3006.zip) |
| | | | | | | |
| :heavy_check_mark: | db817d33 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3006/2/) | 2022-12-06 21:40:17 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221206-213851-db817d-pr3006.zip) |
| :heavy_check_mark: | db817d33 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3006/2/) | 2022-12-06 21:42:33 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221206-213848-db817d-pr3006.aar) |
| :heavy_check_mark: | db817d33 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3006/2/) | 2022-12-06 21:43:37 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221206-213851-db817d-pr3006.zip) |
| | | | | | | |
| :heavy_check_mark: | ce06d017 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3006/3/) | 2022-12-07 19:36:31 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221207-193516-ce06d0-pr3006.zip) |
| :heavy_check_mark: | ce06d017 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3006/3/) | 2022-12-07 19:38:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221207-193511-ce06d0-pr3006.zip) |
| :heavy_check_mark: | ce06d017 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3006/3/) | 2022-12-07 19:38:47 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221207-193511-ce06d0-pr3006.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4e39d7cd | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3006/4/) | 2022-12-09 19:05:52 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221209-190431-4e39d7-pr3006.zip) |
| :heavy_check_mark: | 4e39d7cd | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3006/4/) | 2022-12-09 19:08:48 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221209-190431-4e39d7-pr3006.aar) |
| :heavy_check_mark: | 4e39d7cd | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3006/4/) | 2022-12-09 19:09:22 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221209-190431-4e39d7-pr3006.zip) |
| | | | | | | |
| :heavy_check_mark: | 5588d228 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3006/5/) | 2022-12-09 19:31:39 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221209-193025-5588d2-pr3006.zip) |
| :heavy_check_mark: | 5588d228 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3006/5/) | 2022-12-09 19:33:20 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221209-193025-5588d2-pr3006.zip) |
| :heavy_check_mark: | 5588d228 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3006/5/) | 2022-12-09 19:34:39 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221209-193025-5588d2-pr3006.aar) |

cammellos reviewed (changes requested):
I think this one might leave the chats active, so that you'd be still fetching messages etc (on next login for example).
We have `deactivateChat` that sets the chat to inactive, and that could be used, also I think we should add a few tests to validate the behavior.

jrainville commented:
Good point @cammellos . I added the call to `deactivateChat`. I had to add a new argument so that we can select if we want the history to be cleared. 
I'll add the tests tomorrow morning.

jrainville committed:
fix(community): don't delete chat and messages when leaving community

This creates a smoother experience for users when they leave a community since they can see the exact same messages they had before without having to rely on the mailserver.

jrainville commented:
@cammellos ready for review.
I updated the `TestLeaveAndRejoinCommunity` to check the number of active chats after leave and then after rejoin

jrainville committed:
hotfix: comment out failing test
